sklearn/utils/estimator_checks.py

```diff
@@ -355,10 +358,59 @@ def _is_32bit():
     return struct.calcsize('P') * 8 == 32
 
 
+def _is_pairwise(estimator):
+    """Returns True if estimator has a _pairwise attribute set to True.
+
+    Parameters
+    ----------
+    estimator : object
+        Estimator object to test.
+
+    Returns
+    -------
+    out : bool
+        True if _pairwise is set to True and False otherwise.
+    """
+    return bool(getattr(estimator, "_pairwise", False))
+
+
+def _is_pairwise_metric(estimator):
+    """Returns True if estimator accepts pairwise metric.
+
+    Parameters
+    ----------
+    estimator : object
+        Estimator object to test.
+
+    Returns
+    -------
+    out : bool
+        True if _pairwise is set to True and False otherwise.
+    """
+    metric = getattr(estimator,  "metric", None)
+
+    return bool(metric == 'precomputed')
+
+
+def maybe_pairwise(X, estimator, kernel=linear_kernel):
```

Comment:
I still think this name is a bit too generic. We have a similar-in-spirit function ``multioutput_estimator_convert_y_2d``. Maybe call this one ``pairwise_estimator_convert_X``? Or ``convert_data_pairwise``? I feel ``maybe_pairwise`` doesn't really tell me what the output is going to be.